src/util/compat.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+export opaque type Compatible<T> = [Compat, T];
+type Compat = {|
```

Comment:
Could we rename this to `CompatibilityInfo`/`CompatInfo`, so that we don't have `Compat` and `Compatible` meaning different things, with `toCompat` actually returning a `Compatible`, etc.?

Reply:
Done